### PR TITLE
fix(gradle): identify build folders properly

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**
- Correctly identifies the build folders for a specific build system. 
- Updated unit tests as needed. 

Instead of recursively finding the intersection of parents directories of build files and build folders in a given project which fails when there are multi-path build folders (a/b/c), we first identify the list of directories with build files in a project. We then iterate through this list and check if build folders also exist within them. 

**Why is this change necessary:**
This change properly identifies the project directories with build file (pom.xml, build.gradle) and build folders (target/, build/libs) recursively. 

**How was this change tested:**
- Added UAT for gradle build system with multiple projects. 
 - Added UAT for maven build system with multiple projects. 
 
**Any additional information or context required to review the change:**
Test data needed for multi project gradle and multi project maven build systems is downloaded from s3 bucket. 

**Checklist:**
- [ ] Updated the README if applicable
- [x] Updated or added new unit tests
- [ ] Updated or added new integration tests
- [x] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gdk/static/project_build_system.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
             "build"
         ],
         "build_command_win": [
-            "gradlew",
+            "gradle",
```
Comment (pallavi03) [inline gdk/static/project_build_system.json:23]: why this change?

File: uat/test_uat_build.py [hunk 2]
```diff
@@ -104,3 +107,75 @@ def test_build_template_maven(change_test_dir):
...
+        path_multi_gradle_project,
+        "zip",
+    )
+    os.remove(Path(change_test_dir).joinpath(zip_file))
+    os.chdir(path_multi_gradle_project)
+    assert Path(path_multi_gradle_project).joinpath("recipe.yaml").resolve().exists()
+    config_file = Path(path_multi_gradle_project).joinpath("gdk-config.json").resolve()
+    assert config_file.exists()
+
+    # Update gdk-config file mandatory field like region.
```
Comment (pallavi03) [inline uat/test_uat_build.py:135]: can we use the util method here to update config file?

File: gdk/static/project_build_system.json (hunk 1 above)
Comment (saranyailla) [reply gdk/static/project_build_system.json:23]: just `gradle` works fine without using `.bat` wrapper

File: uat/test_uat_build.py (hunk 2 above)
Comment (saranyailla) [reply uat/test_uat_build.py:135]: util method had extra fields which are not needed here. But updating it for consistency.

**Review by pallavi03 — APPROVED** [review]

**Review by devnaruka — APPROVED** [review]
LGTM

NIT: I would encourage to put given, when and then in comments for test cases to better define our intentions. It helps in increasing code readability. It would help in refactoring these test suites in future.
